### PR TITLE
Update PyTorch 2.2.0, Scilpy 2.0.2, DWI_ML and adapt target SH order based on the training data.

### PR DESCRIPTION
**Pull request by levje**
## PyTorch update to 2.2.0
Didn't change anything except the install.sh file to download another PyTorch version and support different CUDA versions (w.r.t. PyTorch docs). Seems to works fine.

## Scilpy 2.0.2
A few changes here and there to adapt with functions that were moved/removed from scilpy. Logic shouldn't have changed, I used the new functions as they are used in some scilpy scenarios/tests. Changing to scilpy 2.0.2 is useful since its _wheel_ is available on Compute Canada.

## DWI_ML
**IMPORTANT:** Waiting for https://github.com/scil-vital/dwi_ml/pull/234 before merging. Might have to adapt the commit hash in the requirements.txt to refer to the main once that PR is merged.

Update of DWI_ML is useful because it now uses PyTorch 2.2.0 and Scilpy 2.0.2 which is needed for Compute Canada.

## SH Order changes:
During training:
- When loading the first subject to initialize the environment, use the input_volume from that subject to extract sh_order that the model will be trained on.
- Save the sh_target_order to the hyperparameters.json file.
- I adapted the default example model's hyperparameters.json file accordingly.

During tracking/testing (with ttl_track*):
- Load the sh_target_order directly from the hyperparameters.json file. This way the networks are initialized with the input size corresponding to the one from the pretrained model.

Other:
- Also removed `self.interface_seeding = hyperparams['interface_seeding']` and `self.add_neighborhood = hyperparams['add_neighborhood']` since they didn't seem to be used anymore and they caused issues when training/tracking.

### CODE REVIEW

File: requirements.txt
Comment (levje) [inline requirements.txt]: Need to update that once that linked branch is merged.

File: requirements.txt
Comment (levje) [inline requirements.txt]: Might need to change h5py version if it's changed in DWI & Scilpy

**Comment by AntoineTheb** [comment]
Looks good ! I'll do a proper review once I'm back. Ideally, we would set the option added in this PR: https://github.com/scil-vital/dwi_ml/pull/238 to False as I realised when profiling that cache clearing slows down tracking **a lot**. I'll try to finish this PR asap.

**Comment by levje** [comment]
Checks will fail, but once the clearing cache PR in DWI-ML is merged, we should be good to merge that one if everything is ok. Also, make sure that the clearing_cache option is merged into the for_beluga_scilpy2 branch.

**Review by AntoineTheb — APPROVED** [review]

**Pull request merged**
